modules/dnn/CMakeLists.txt

```diff
@@ -162,6 +167,11 @@ if(HAVE_TIMVX)
     list(APPEND libs -Wl,--whole-archive ${TIMVX_LIBRARY} -Wl,--no-whole-archive)
 endif()
 
+if(HAVE_CANN)
+  list(APPEND include_dirs ${CANN_INCLUDE_DIRS})
+  list(APPEND libs -Wl,--whole-archive ${CANN_LIBRARIES} -Wl,--no-whole-archive)
+endif()
```

Comment:
Need to reuse `ocv_add_external_target()` approach for 3rdparty libraries.

Reply:
How about keep it as-is since other 3rdparty libraries are still using this approach? We can do this kind of change for all other 3rdparty libraries in a separate pr.

Reply:
Do we still need this? (as there is `ocv.3rdparty.cann` usage below)

Reply:
Yes, we do. `ocv.3rdparty.cann` is used to link for `opencv_test_dnn` only for now.